src/core/qgssqliteexpressioncompiler.cpp

```diff
@@ -90,11 +90,41 @@ QString QgsSQLiteExpressionCompiler::sqlFunctionFromFunctionName( const QString
     { "round", "round" },
     { "trim", "trim" },
     { "upper", "upper" },
+    { "make_datetime", "date" },
+    { "make_date", "date" },
+    { "make_time", "date" },
   };
 
   return FN_NAMES.value( fnName, QString() );
 }
 
+QStringList QgsSQLiteExpressionCompiler::sqlArgumentsFromFunctionName( const QString &fnName, const QStringList &fnArgs ) const
+{
+  QStringList args( fnArgs );
+  if ( fnName == QLatin1String( "make_datetime" ) )
+  {
+    args = QStringList( QStringLiteral( "DATE('%1-%2-%3 %4:%5:%6')" ).arg( args[0].rightJustified( 4, '0' ) )
+                        .arg( args[1].rightJustified( 2, '0' ) )
```

Comment:
There's an issue here -- make_date etc don't necessarily have numeric arguments. You could potentially call them with sub expressions or field references

Reply:
And the expression engine doesn't resolve those nodes first?

Reply:
Our expression engine does, but the compiler runs before all this.

I think what we need to do is check if the node values used as arguments are constant, and if they are then it's safe to use your approach. If not the compilation should fail.

Reply:
And how is that done sir? :)